### PR TITLE
refactor: #205 Setup eslint style

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:
- **Purpose**: Provided consistency with the use of quotes, semicolons, etc.
- **Approach**: Setup eslint style as a complement to eslint

**Type of change**

- [x] Refactor

**Updates**

- Installs stylistic eslint and ts plugins.
- Configure eslint.
- apply code changes.

stylistic-eslint and prettier can "disagree" sometimes, that's why some rules have been disabled.

### Test plan :test_tube:

- Code remains functionally the same, GitHub CI should run without complains

### Author to check :eyeglasses:
- [x] Project and all contained modules builds successfully
- [ ] Self-/dev-tested 
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,18 +33,17 @@
...
     "//eslint": "performs static analysis over project files and examples",
     "eslint": "eslint --fix lib/*.ts test/*.js examples/**/*.js",
-
     "//prettier": "performs code formatting over project files and examples",
     "prettier": "prettier --write lib/*.ts test/*.js examples/**/*.js",
-
     "//prepare": "prepare project for distribution",
     "prepare": "tsc",
-
     "//test": "runs tests on main project",
```
Comment (miquelbeltran) [inline package.json:40]: I think the blank lines were removed by the npm tool when I ran `npm -i -D ....`

**Comment by TheRealAgentK** [comment]
I don't see why quotes would be preferred over backticks apart from the AirBnb style guide saying so, to be honest.

Many yrs ago, backticks used to be slower, but essentially seem to have been offering the same performance for a long time. They're more flexible and support multi-line and kind-of represent modern JS.

**Comment by TheRealAgentK** [comment]
> I don't have a preference, but I'd prefer a consistent style that is enforced by the static analysis tools.

Fully agreeing!

> Using backticks everywhere in the codebase looks like a bit too much to me, but it is doable automatically, we only need to change the configuration of the rule https://eslint.org/docs/latest/rules/quotes to be `["error", "backtick"]`.

^ This is the only point of discussion, really. 

Here's one list of pros/cons: 

https://medium.com/@sathiranipun/unveiling-the-magic-of-backticks-single-quotes-and-double-quotes-in-javascript-9ef33d32b9fe

Someone looking into the performance (in a browser though): https://dev.to/efpage/the-fantastic-speed-of-template-literals-4f0p

There are obviously numerous punters with opinions out there, but I don't see the drawback of using backticks exclusively at this stage. They're more flexible and they are the more modern style of doing things.

Note: this is *not* the hill I'm going to die on, but I'm curious to what others think. :smile: 

> There is another option, which is to configure the rule as this: `["error", "double", { "allowTemplateLiterals": true }]` this allows you to either chose with double quotes or backticks freely. See https://eslint.org/docs/latest/rules/quotes#allowtemplateliterals
> 
> Alternatively, we can completely disable this rule. But I'd prefer to at least have the above to enforce either double quotes or backticks.
> 
> What do you think @TheRealAgentK ?

I don't want to over-complicate things, let's not complicate rules and just make a decision on either consistent approach.

**Comment by miquelbeltran** [comment]
The good thing about eslint is that running the alias `npm run eslint` will apply fixes automatically (as it has the `--fix` param) so will change any wrong doings.

I will play a bit with the rule config to see if setting backticks is doable and how it will look like.

**Comment by miquelbeltran** [comment]
Another thing to keep in mind is that backticks cannot be used for property values, so we will have to define another rule if we want property values as single or double quotes, e.g. 
```js
var obj = { 'prop-name': `value` };
```

**Review by TheRealAgentK — APPROVED** [review]
LGTM after an internal discussion on how we want to handle this.

**Review by sumitramanga — APPROVED** [review]
For visibility, I'm happy with this too :)

**Pull request merged**